src/urdf_parser_py/xml_reflection/basics.py

```diff
@@ -1,14 +1,21 @@
 import string
 import yaml
 import collections
-from lxml import etree
+
+from xml.etree import ElementTree as ET
+from xml.dom import minidom
+
 
 def xml_string(rootXml, addHeader=True):
-    # Meh
-    xmlString = etree.tostring(rootXml, pretty_print=True, encoding='unicode')
-    if addHeader:
-        xmlString = '<?xml version="1.0"?>\n' + xmlString
-    return xmlString
+    # From: https://stackoverflow.com/a/1206856/170413
+    # TODO(eacousineau): This does not preserve attribute order. Fix it.
+    dom = minidom.parseString(ET.tostring(rootXml))
+    xml_string = ""
+    lines = dom.toprettyxml(indent="  ").split("\n")
```

Comment:
I had two major [comments](https://github.com/ros/urdf_parser_py/pull/32#pullrequestreview-135819431) on the original PR.

The first one was about pretty printing the output.  That one isn't super-important, but was something that the old one did.  It looks like this one is doing that as well, which is good, and takes care of that point.

The second point was about preserving XML comments, which lxml does, but minidom does not.  I guess we have to answer whether that is an important use case, and whether we can make that happen with minidom.  I don't know that answer to either one.

Reply:
It looks like the `melodic-devel` branch is explicitly filtering XML comments out. Is there another API in this repo that gives access to them?

https://github.com/ros/urdf_parser_py/blob/aa7199facecd1227409fb9858eb8ccd90fa8716e/src/urdf_parser_py/xml_reflection/basics.py#L34-L39

Reply:
I was originally worried about round-tripping through this API, like:

```
from urdf_parser_py.urdf import URDF

with open('robot.urdf', 'r') as infp:
    robot = URDF.from_xml_string(infp.read())

print(robot.to_xml_string())
```

But you are right, we are already stripping out those comments.  Personally I think that is a bug, but that is orthogonal to this PR.  So I think we can proceed.